lib/bridge.js

```diff
@@ -237,11 +240,109 @@ Bridge.prototype.run = function(port, config, appServiceInstance) {
         }
         self.opts.controller.onLog(line, false);
     });
+    this.customiseAppservice(this.appService);
     this.appService.listen(port);
     return this.loadDatabases();
 };
 
 /**
+ * Apply any customisations required on the appService object.
+ * Subclasses are intended to override this method where appropriate. If
+ * overridden the subclass method will still have to call the one in this
+ * class.
+ * @param {AppService} the AppService instance of the object
+ */
+Bridge.prototype.customiseAppservice = function(appService) {
+    // TODO(paul): This is gut-wrenching into the AppService instance itself.
+    //   Maybe an API on that object would be good?
+    var app = appService.app;
+
+    if (this.opts.controller.thirdPartyLookup) {
+        var lookupController = this.opts.controller.thirdPartyLookup;
+
+        var protocolFilter;
+        if ("protocols" in lookupController) {
```

Comment:
`if (lookupController.protocols) {` is a better way of saying this because it is a static property name.

It looks like you're setting up a map with which to filter protocols. Given how few protocols we expect to be present (1 typically, maybe 2?), I would just linear scan and keep it as it is (a string array). This means you can remove `protocolFilter` and just replace checks with:

```javascript
if (lookupController.protocols.indexOf($PROTOCOL) === -1) {
    // not present
}
```
And do `lookupController.protocols.length === 0` to see if it was specified in the first place. You will need to default initialise it though in order to make sure that `protocols` is defined:
```javascript
lookupController.protocols = lookupController.protocols || [];
```